pandas/tests/tseries/offsets/test_offsets.py

```diff
@@ -870,3 +873,17 @@ def test_dateoffset_immutable(attribute):
     msg = "DateOffset objects are immutable"
     with pytest.raises(AttributeError, match=msg):
         setattr(offset, attribute, 5)
+
+
+@pytest.mark.parametrize(
+    "offset_type",
+    [
+        "QuarterBegin",
+        "QuarterEnd",
+        "BQuarterBegin",
+        "BQuarterEnd",
+    ],
+)
+def test_startingMonth_deprecation_warning(offset_type):
```

Comment:
FutureWarning and add the issue number as a comment

Reply:
Done.